text_classification.py

```diff
@@ -0,0 +1,487 @@
+                train.csv file with split_ratio percent.
+            - train_data: a datset objects based on test.csv file.
+            - valid_data: a datset objects based on
+                train.csv file with 1-split_ratio percent.
+
+        """
+        rnd = dataset.RandomShuffler(random_state)
+        train_ratio, test_ratio, valid_ratio = split_ratio, 0.0, 1 - split_ratio
+        train_examples, test_examples, valid_examples = \
+            dataset.rationed_split(self.train_examples,
+                                   train_ratio, test_ratio, valid_ratio, rnd)
+        test_examples = self.test_examples
+        split = tuple(TextDataset(d, self.fields)
+                      for d in (train_examples, test_examples, valid_examples)
+                      if d is not None)
+        for subset in split:
+            subset.sort_key = self.sort_key
+
+        return split
+
+    def iters(self, split_ratio=0.7, batch_size=32, device='cpu', random_state=None):
+        """Create iterator objects for splits of the dataset.
+
+        Arguments:
+            split_ratio: split train_examples into train set (split_ratio)
+                and valid set (1-split_ratio). Default: 0.7
+            batch_size: batch size. Default: 32
+            device: the device to sent data. Default: 'cpu'
+            random_state: the random state provided by user. Default: None
+
+        Examples:
+            >>> train_iter, test_iter, valid_iter = txt_cls.iters(device="cpu")
+
+        Outputs:
+            - train_iter: a iterator based on
+                train.csv file with split_ratio percent.
+            - train_iter: a iterator based on test.csv file.
+            - valid_iter: a iterator based on
+                train.csv file with 1-split_ratio percent.
+
+        """
+
+        train, test, valid = self.split(split_ratio, random_state)
+        return generate_iterators(
+            (train, test, valid), [batch_size] * 3, device=device)
+
+    def download(self):
+        """Download the dataset if it doesn't exist in processed_folder already."""
+
+        train_csv_path = os.path.join(self.raw_folder,
+                                      self.dataset_name + '_csv',
+                                      'train.csv')
+        test_csv_path = os.path.join(self.raw_folder,
+                                     self.dataset_name + '_csv',
+                                     'test.csv')
+        if os.path.exists(train_csv_path) and os.path.exists(test_csv_path):
+            return
+
+        os.makedirs(self.raw_folder)
+        filename = self.dataset_name + '_csv.tar.gz'
+        url = self.url
+        path = os.path.join(self.raw_folder, filename)
+        download_from_url(url, path)
+        extract_archive(path, self.raw_folder, remove_finished=True)
+
+        print('Dataset %s downloaded.' % self.dataset_name)
+
+    def preprocess(self, raw_folder=None, processed_folder=None):
```

Comment:
This seems useful in its own right, i.e. it applies a preprocessing function to a set of text files. Can we move it out?

Reply:
Similar to download, we don't need those wrappers inside the class.